analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -72,52 +72,47 @@ private static void CheckForIssue(SyntaxNodeAnalysisContext analysisContext)
                 return;
             }
 
-            var parameterNameValue = default(Optional<object>);
-            var messageValue = default(Optional<object>);
-            for (var i = 0; i < methodSymbol.Parameters.Length; i++)
-            {
-                var argument = objectCreation.ArgumentList.Arguments[i];
-                var argumentExpression = objectCreation.ArgumentList.Arguments[i].Expression;
-                RetrieveParameterAndMessageArgumentValue(
-                    argument.NameColon != null
-                        ? argument.NameColon.Name.Identifier.ValueText
-                        : methodSymbol.Parameters[i].MetadataName,
-                    argumentExpression,
-                    analysisContext.SemanticModel,
-                    ref parameterNameValue,
-                    ref messageValue);
-            }
+            var parameterAndMessage = RetrieveParameterAndMessageArgumentValue(methodSymbol, objectCreation, analysisContext.SemanticModel);
 
-            if (!parameterNameValue.HasValue)
+            if (!parameterAndMessage.Item1.HasValue)
             {
                 // can't check non-constant strings OR argument is not set
                 return;
             }
 
             var methodArgumentNames = GetMethodArgumentNames(objectCreation.Expression).ToHashSet();
-            if (!methodArgumentNames.Contains(TakeOnlyBeforeDot(parameterNameValue)))
+            if (!methodArgumentNames.Contains(TakeOnlyBeforeDot(parameterAndMessage.Item1)))
             {
-                var message = messageValue.HasValue && messageValue.Value != null && methodArgumentNames.Contains(TakeOnlyBeforeDot(messageValue))
+                var message = parameterAndMessage.Item2.HasValue && parameterAndMessage.Item2.Value != null && methodArgumentNames.Contains(TakeOnlyBeforeDot(parameterAndMessage.Item2))
                     ? ConstructorParametersInverted
-                    : string.Format(InvalidParameterName, parameterNameValue.Value);
+                    : string.Format(InvalidParameterName, parameterAndMessage.Item1.Value);
```

Comment:
I am confused here because from what I see, both `parameter` and `message` are used in constructing the message, right?

Also, `methodArgumentNames.Contains(TakeOnlyBeforeDot(parameterAndMessage.Item2))`  - we look for the `messageValue` inside the `methodArgumentsName` 

Actually, in the end the `messageArgument` (previously `messageValue` ) isn't used at all in the message


would it be possible to take a moment and find better names for this logic?